src/map/Map.tsx

```diff
@@ -1,41 +1,290 @@
+    highlightedPathDetailSegments: Coordinate[][]
+}
+
+export default function({
+                            viewport,
+                            selectedPath,
+                            paths,
+                            queryPoints,
+                            mapStyle,
+                            pathDetailPoint,
+                            highlightedPathDetailSegments
+                        }: MapProps) {
+    const [popupCoordinate, setPopupCoordinate] = useState<Coordinate | null>(null)
+    const currentPaths = paths
+        .map((path, i) => {
+            return {
+                path,
+                index: i
+            }
+        })
+        .filter(indexPath => indexPath.path !== selectedPath)
+    return <ReactMapGL
+        mapStyle={getStyle(mapStyle)}
+        {...viewport}
+        width='100%'
+        height='100%'
+        mapOptions={{
+            // todo: maxBounds
+            renderWorldCopies: false
+        }}
+        onLoad={() => Dispatcher.dispatch(new MapIsLoaded())}
+        onViewportChange={(nextViewport: ViewportStoreState) => Dispatcher.dispatch(new SetViewport(nextViewport))}
+        // todo: minor glitch: when we hover the map before the path got loaded we get an error in the console
+        interactiveLayerIds={currentPaths.length === 0 ? [] : [pathsLayerKey]}
+        onClick={(e) => {
+            const feature = e.features?.[0]
+            if (feature) {
+                // select an alternative path if clicked
+                if (feature.layer.id === pathsLayerKey) {
+                    const index = feature.properties!.index
+                    const path = currentPaths.find(indexPath => indexPath.index === index)
+                    Dispatcher.dispatch(new SetSelectedPath(path!.path))
+                }
+            }
+        }}
+        onContextMenu={(e) => {
+            e.preventDefault()
+            setPopupCoordinate({ lng: e.lngLat[0], lat: e.lngLat[1] })
+        }}
+        // todo: long touch handler
+        // const handler = new LongTouchHandler(e => this.popup.show(e.lngLat));
+        // onTouchStart={handler.onTouchStart}
+        // onTouchEnd={handler.onTouchEnd}
+        // onTouchMove={handler.onTouchEnd}
+    >
+        {popupCoordinate && <Popup
+            longitude={popupCoordinate.lng}
+            latitude={popupCoordinate.lat}
+            closeOnClick={true}
+            closeButton={false}
+            // todo
+            // closeOnMove: true,
+        >
+            <PopupComponent
+                coordinate={popupCoordinate}
+                queryPoints={queryPoints}
+                onSelect={() => setPopupCoordinate(null)} />
+        </Popup>}
+        {createQueryPointMarkers(queryPoints)}
+        {pathDetailPoint && createPathDetailMarker(pathDetailPoint)}
+        {createUnselectedPaths(currentPaths)}
+        {createSelectedPath(selectedPath)}
+        {createHighlightedPathSegments(highlightedPathDetailSegments)}
```

Comment:
In principle we could replace this with something like `{...layers}` and the `layers` would be passed into the map component as prop. This way we could just add different layers to the map without modifying the `Map` component. Then again, I wonder what there would be really left for the `Map` component to do. The `react-map-gl` kind of *is* our map component already... One important question might be what should be re-used in case for example we wanted to 'switch' the app to isochrones or map-matching.